File: Java/src/TwoSumII.java
import java.util.Arrays;


// Find pair of numbers that sum to a target (input array is sorted)

public class TwoSumII {
    public static void main(String[] args) {
        int[] nums = {1, 2, 4, 5, 7, 8, 9};
        System.out.println(Arrays.toString(solve(nums, 7)));
    }

    static int[] solve(int[] nums, int target) {
        int i = 0, j = nums.length-1;

        while (i < j) {
            int sum = nums[i] + nums[j];
            if (sum < target) i++;
            if (sum > target) j--;
            if (sum == target) return new int[] {i, j};
        }
        return new int[] {-1, -1};
    }
}

File: Python/two_sum_ii.py
# find the pairs which sum upto the target when the input array is sorted
def solve(nums, target):
    i, j = 0, len(nums) - 1
    while i < j:
        _sum = nums[i] + nums[j]
        if _sum < target:
            i += 1
        if _sum > target:
            j -= 1
        if _sum == target:
            return [i, j]
    return [-1, -1]


def main():
    arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
    print(solve(arr, 50))  # [5, 8]


if __name__ == '__main__':
    main()
